ParticleObjects/Source/surfaceSource_class.f90

```diff
@@ -0,0 +1,324 @@
+module surfaceSource_class
+
+  use numPrecision
+  use universalVariables
+  use genericProcedures,     only : fatalError
+  use particle_class,        only : particleState, P_NEUTRON, P_PHOTON
+  use particleDungeon_class, only : particleDungeon
+  use dictionary_class,      only : dictionary
+  use configSource_inter,    only : configSource, kill_super => kill
+  use geometry_inter,        only : geometry
+  use RNG_class,             only : RNG
+
+  implicit none
+  private
+
+  !!
+  !! Generates a source representing a black body surface
+  !! Put together quite quickly so very specific in use and not perfect
+  !!   - Currently only allows a circle or square aligned on x y or z axis, with
+  !!     a certain radius or side length
+  !!   - Requires deltat and nParticles in input file to be the same as specified elsewhere
+  !!     in file, can change to not require these inputs with some more thought
+  !!   - May still contain unnecessary lines of code copied from pointSource_class.f90
+  !!
+  !! Private members:
+  !!   r            -> source position
+  !!   dir          -> optional source direction
+  !!   particleType -> source particle type
+  !!   isMG         -> is the source multi-group?
+  !!   isIsotropic  -> is the source isotropic?
+  !!
+  !! Interface:
+  !!   init              -> initialise point source
+  !!   sampleType        -> set particle type
+  !!   samplePosition    -> set particle position
+  !!   sampleEnergy      -> set particle energy
+  !!   sampleEnergyAngle -> sample particle angle
+  !!   kill              -> terminate source
+  !!
+  !! Sample Dictionary Input:
+  !!   source {
+  !!       type surfaceSource;
+  !!       shape circle    ! circle or square;
+  !!       size 5;         ! radius(circle) or side length(square)
+  !!       axis x;         ! axis normal to planar shape
+  !!       pos 0;          ! distance along axis to place plane
+  !!       T 1;            ! temperature of source boundary
+  !!       nParticles 100; ! Number of particles emitted per time step, for now has to be
```

Comment:
Can this not be derived from an intensity? Or shouldn't there be an intensity in the input?